sdk/core/core-client/src/serializer.ts

```diff
@@ -954,10 +954,16 @@ function deserializeCompositeType(
       let propertyInstance;
       let res = responseBody;
       // traversing the object step by step.
+      let steps = 0;
       for (const item of paths) {
         if (!res) break;
+        steps++;
         res = res[item];
       }
+      // only accept null when reaching the last position of object otherwise it would be undefined
+      if (res === null && steps < paths.length) {
```

Comment:
does <= make any difference?

Reply:
yes, it would cause the res changed from null to undefined.